dali/pipeline/operators/crop/crop.cu

```diff
@@ -116,17 +116,15 @@ void Crop<GPUBackend>::ValidateHelper(TensorList<GPUBackend> *output) {
 template<>
 void Crop<GPUBackend>::SetupSharedSampleParams(DeviceWorkspace *ws) {
   const auto & input = ws->Input<GPUBackend>(0);
-
-  if (output_type_ == DALI_NO_TYPE)
-    output_type_ = input.type().id();
+  CastPermuteAttr::SetupSharedSampleParams(ws);
```

Comment:
Why Crop has always to call CastPermuteAttrs? That doesn'ts seem ok

Reply:
When `output_type_` of the operator is not defined (its value equal to DALI_NO_TYPE), it should be the same as the type of the input data. It is why following methods:
```
  void CastPermuteAttr::SetupSharedSampleParams(const DeviceWorkspace *ws) {
    if (output_type_ == DALI_NO_TYPE)
      output_type_ = ws->Input<GPUBackend>(0).type().id();
  }

  void CastPermuteAttr::SetupSharedSampleParams(const SampleWorkspace *ws) {
    if (output_type_ == DALI_NO_TYPE)
      output_type_ = ws->Input<CPUBackend>(0).type().id();
  }
```
are called. 
Yes, `output_type_` is not used in the `Crop`, but we need to pass it to next operator, where it could be used.